libraries/Microsoft.Bot.Builder.Azure/Queues/ContinueConversationLater.cs

```diff
@@ -0,0 +1,144 @@
+﻿// Licensed under the MIT License.
+// Copyright (c) Microsoft Corporation. All rights reserved.
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using AdaptiveExpressions.Properties;
+using Azure.Storage.Queues;
+using Microsoft.Bot.Builder.Dialogs;
+using Newtonsoft.Json;
+
+namespace Microsoft.Bot.Builder.Azure.Queues
+{
+    /// <summary>
+    /// Action which schedules a conversation to be continued later by writing a EventActivity(Name=ContinueConversation) to a Azure Storage queue.
+    /// </summary>
+    /// <remarks>
+    /// This class works by writing an EventActivity(Name=ConversationUpdate) to an azure storage queue with visibility policy to 
+    /// make it visible at a future point in time. 
+    /// 
+    /// The queue needs a process (such as a webjob/azure function) monitoring incoming activities and processing them by either:
+    ///   - posting the activity back to the bot itself via BotFrameworkHttpClient.PostActivityAsync(botId, botEndpoint, activity).
+    /// OR
+    ///   - processing the activity directly via adapter.ProcessActivity(activity, ...); 
+    ///     NOTE: adapter.ProcessActivity() understands that EventActivity(Name=ConversationUpdate) should be processed as ContinueConversation() pipeline.
+    /// 
+    /// This dialog returns the reciept information for the queued activity as the result of the dialog.
+    /// </remarks>
+    public class ContinueConversationLater : Dialog
```

Comment:
I think this Dialog should probably leave in the Dialogs package, then we can create primitives to deal with Azure queues in the Azure package and we should create some abstraction like we did with IStorage so we are not tightly coupled to Azure Queues.